core/src/main/java/com/zextras/carbonio/user_management/services/UserService.java

```diff
@@ -45,29 +49,55 @@ private UserInfo createUserInfo(GetAccountInfoResponse accountInfo) {
     return userInfo;
   }
 
+  public Response getUsers(List<String> userIds, String token) {
+    return Response.ok().entity(
+      (new HashSet<>(userIds)).stream().map(userId -> {
```

Comment:
Instead of creating a new hashset data structure where you copy the ids, it's easier to just userIds.stream().distinct().map(...) on the original list